jsapp/js/api.ts

```diff
@@ -33,20 +34,24 @@ export function handleApiFail(response: FailResponse) {
     message = htmlDoc.getElementsByClassName('errormsg')?.[0]?.innerHTML;
   }
 
-  if (!message) {
-    message = t('An error occurred');
+  const htmlMessage = message;
+
+  if (toastMessage || !message) {
+    message = toastMessage || t('An error occurred');
     if (response.status || response.statusText) {
-      message += ` — ${response.status} ${response.statusText}`;
+      message += `\n\n${response.status} ${response.statusText}`;
     } else if (!window.navigator.onLine) {
       // another general case — the original fetch response.message might have
       // something more useful to say.
-      message += ' — ' + t('Your connection is offline');
+      message += '\n\n' + t('Your connection is offline');
     }
   }
 
+  // show the prettiest version of the error to the user
   notify.error(message);
 
-  window.Raven?.captureMessage(message);
+  // prefer sending the HTML error to Raven, since it should contain more detailed information
+  window.Raven?.captureMessage(htmlMessage || message);
```

Comment:
Kind of the same gripe about variable naming above, and more of a comment about the human-readable names and comments vs. the logic itself: the "HTML error" here "works for Werkzeug Debugger only" according to the comment on lines 29-30. That only runs locally, and so we would not be interested in sending those messages to Raven → Sentry/GlitchTip, but rather showing them immediately to the front-end developer.

If I'm reading the logic right, though, `htmlMessage` actually also would include an unparsed, plain-text version of the server's response, and so that would perhaps have more helpful information, even in production, to send via Raven :ok_hand: 

(And it does seem that `notify.error(message)`  would show the unpretty parsed-from-HTML message from Werkzeug / `manage.py runserver` in the notification for the developer's benefit, following the seeming intent of the original code :+1: It'd only be hidden if a `toastMessage` were specified, right? Again, I think the logic's fine—it's just an unclear naming thing for me.)

Thanks :)